src/client/Microsoft.Identity.Client/ConfidentialClientApplication.cs

```diff
@@ -77,6 +77,35 @@ public sealed partial class ConfidentialClientApplication
                 authorizationCode);
         }
 
+        /// <summary>
+        /// Acquires a security token from the authority configured in the app using the authorization code
+        /// previously received from the STS.
+        /// It uses the OAuth 2.0 authorization code flow (See https://aka.ms/msal-net-authorization-code).
+        /// It's usually used in web apps (for instance ASP.NET / ASP.NET Core web apps) which sign-in users,
+        /// and can request an authorization code.
+        /// This method does not lookup the token cache, but stores the result in it, so it can be looked up
+        /// using other methods such as <see cref="IClientApplicationBase.AcquireTokenSilent(IEnumerable{string}, IAccount)"/>.
+        /// </summary>
+        /// <param name="scopes">Scopes requested to access a protected API</param>
+        /// <param name="authorizationCode">The authorization code received from the service authorization endpoint.</param>
+        /// <param name="codeVerifier">A dynamically created cryptographically random key used to provide proof of possession for the <paramref name="authorizationCode"/>.</param>
+        /// <returns>A builder enabling you to add optional parameters before executing the token request</returns>
+        /// <remarks>You can set optional parameters by chaining the builder with:
+        /// <see cref="AbstractAcquireTokenParameterBuilder{T}.WithAuthority(string, bool)"/>,
+        /// <see cref="AbstractAcquireTokenParameterBuilder{T}.WithExtraQueryParameters(Dictionary{string, string})"/>,
+        /// </remarks>
+        public AcquireTokenByAuthorizationCodeParameterBuilder AcquireTokenByAuthorizationCode(
+            IEnumerable<string> scopes,
+            string authorizationCode,
+            string codeVerifier)
```

Comment:
I do not think we should make this parameter mandatory. It should only exists in the `.WithCodeVerifier` method

Reply:
it could be null by default

Reply:
Agree with @jmprieur , I think that's what we agreed upon. but it should be null, and then you can just have one method.

---
In reply to: [609821050](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2533#discussion_r609821050) [](ancestors = 609821050)